Firebase/Auth/Source/Public/FIRAuthDataResult.h

```diff
@@ -38,14 +37,7 @@ NS_SWIFT_NAME(AuthDataResult)
 /** @property user
     @brief The signed in user.
  */
-@property(nonatomic, readonly, nullable) FIRUser *user;
-
-/** @property credential
-    @brief The updated OAuth credential after the the sign-in, link and reauthenticate action.
-    @detial This property is for OAuth sign in only.
- */
-@property(nonatomic, readonly, nullable) FIROAuthCredential *credential;
-
+@property(nonatomic, readonly) FIRUser *user;
```

Comment:
Isn't making `user` nullable a bugfix that should stay?

Reply:
See below and the message of the 3rd commit for my explanation and please let me know what do you believe would be the better way to resolve this.